# HG changeset patch
# User Rik <rik@octave.org>
# Date 1388769364 28800
#      Fri Jan 03 09:16:04 2014 -0800
# Node ID 0ecd4618b1fc716f84ec922060c65019ddf660bc
# Parent  b7d3e1f762e12f728a359b67ab878b64aa9ee228
voronoi.m: Fix input validation so it doesn't open blank figure window.

voronoi.m: Fix input validation so it doesn't open blank figure window.

diff --git a/scripts/geometry/voronoi.m b/scripts/geometry/voronoi.m
--- a/scripts/geometry/voronoi.m
+++ b/scripts/geometry/voronoi.m
@@ -69,24 +69,23 @@
 
 function [vx, vy] = voronoi (varargin)
 
   if (nargin < 1)
     print_usage ();
   endif
 
   narg = 1;
+  hax = NaN;
   if (isscalar (varargin{1}) && ishandle (varargin{1}))
     hax = varargin{1};
     if (! isaxes (hax))
       error ("voronoi: HAX argument must be an axes object");
     endif
     narg++;
-  elseif (nargout < 2)
-    hax = gca ();
   endif
 
   if (nargin < 1 + narg || nargin > 3 + narg)
     print_usage ();
   endif
 
   x = varargin{narg++};
   y = varargin{narg++};
@@ -154,16 +153,19 @@ function [vx, vy] = voronoi (varargin)
                 ismember (edges(2, :), poutside);
   edges(:, edgeoutside) = [];
 
   ## Get points of the diagram
   Vvx = reshape (p(edges, 1), size (edges));
   Vvy = reshape (p(edges, 2), size (edges));
 
   if (nargout < 2)
+    if (isnan (hax))
+      hax = gca ();
+    endif
     h = plot (hax, Vvx, Vvy, linespec{:}, x, y, '+');
     lim = [xmin, xmax, ymin, ymax];
     axis (lim + 0.1 * [[-1, 1] * xdelta, [-1, 1] * ydelta]);
     if (nargout == 1)
       vx = h;
     endif
   else
     vx = Vvx;
